packages/apollo-engine-reporting/src/agent.ts

```diff
@@ -226,13 +231,24 @@ export class EngineReportingAgent<TContext = any> {
 
   public constructor(options: EngineReportingOptions<TContext> = {}) {
     this.options = options;
-    this.apiKey = options.apiKey || process.env.ENGINE_API_KEY || '';
+    const legacyApiKeyFromEnv = process.env.ENGINE_API_KEY;
+    const apiKeyFromEnv = process.env.APOLLO_KEY;
+
+    if(legacyApiKeyFromEnv && apiKeyFromEnv) {
+      throw new Error(`Cannot set both ${legacyKeyEnvVar} and ${keyEnvVar}. Please only set ${keyEnvVar}`);
+    }
+
+    this.apiKey = options.apiKey || apiKeyFromEnv || legacyApiKeyFromEnv || '';
```

Comment:
Possibly out of scope, but I think we should add a warning here as well if they're using both `options.apiKey` in tandem with either of the env vars.

```js
if (options.apiKey && (apiKeyFromEnv || legacyApiKeyFromEnv)) {
  this.logger.warn(
    'An API key was found in both the server config and an ' +
    'environment variable. Apollo Server will use the key provided ' +
    'via the configuration object.'
  );
}
```

Reply:
Hm. Why do we consider this "warn"-able? I suppose I always assumed that was a reasonable usage for supporting behavior of "by default, do this, but I can inject config to override" 🤷‍♂ 

I suppose it might happen by accident, though!

Reply:
It doesn't have to be necessarily be a `warn`, I think `info`/`debug` could be sufficient. I don't think running with both set should be the norm though - maybe a local dev scenario. In which case, I'd appreciate being notified which key is being used in the event that more than one is set. I'm more concerned about the accident!